integrations/snowflake/src/haystack_integrations/components/retrievers/snowflake/auth.py

```
@@ -0,0 +1,283 @@
+        """
+        Gets the password for URI construction in SNOWFLAKE authentication.
+
+        :returns: Resolved password value or None.
+        :raises ValueError: If secret resolution fails.
+        """
+        if self.authenticator == "SNOWFLAKE" and self.api_key:
+            return self.resolve_secret_value(self.api_key)
+        return None
+
+    def create_masked_params(self, params: list) -> list:
+        """
+        Creates a masked version of authentication parameters for safe logging.
+
+        :param params: Original parameter list.
+        :returns: Parameter list with sensitive information masked.
+        """
+        masked_params = []
+
+        for param in params:
+            masked_param = param
+
+            # Mask private key password
+            if self.private_key_file_pwd:
+                private_key_pwd = self.resolve_secret_value(self.private_key_file_pwd)
+                if private_key_pwd and private_key_pwd in param:
+                    masked_param = param.replace(private_key_pwd, "***REDACTED***")
+
+            # Mask OAuth client secret
+            if self.oauth_client_secret:
+                client_secret = self.resolve_secret_value(self.oauth_client_secret)
+                if client_secret and client_secret in param:
+                    masked_param = masked_param.replace(client_secret, "***REDACTED***")
+
+            masked_params.append(masked_param)
+
+        return masked_params
+
+    def test_connection(self, user: str, account: str, database: Optional[str] = None) -> bool:
+        """
+        Tests the connection with the provided credentials.
+
+        :param user: Snowflake username.
+        :param account: Snowflake account identifier.
+        :param database: Optional database name.
+        :returns: True if connection is successful, False otherwise.
+        """
+        try:
+            try:
+                import snowflake.connector  # type: ignore[import-not-found]  # noqa: PLC0415
```

Similarly, I would import this at the top and let the code fail when it's not installed.